src/script.rs

```diff
@@ -595,10 +547,11 @@ impl Script<'_> {
         let mut pc = self.0;
         let mut last_opcode = Opcode::Operation(OP_INVALIDOPCODE);
         while !pc.is_empty() {
-            let opcode = match Self::get_op(&mut pc) {
+            let (opcode, new_pc) = match Self::get_op(pc) {
                 Ok(o) => o,
                 Err(_) => break,
```

Comment:
nonblocking: It's annoying that the error case here is ignored in the original C++ code. It will probably be worth a comment here, since once we transition fully to the Rust implementation this behavior will be puzzling.

Reply:
Yeah … I can only guess that this is done to prevent someone from creating script that’s too expensive, but then doing something invalid at the end to prevent this check from catching it. _But_
1. the interpreter also tracks op_count, so it at least wouldn’t get past 201 no matter what; and
2. if this _did_ error, then we’d know not to bother running the script. 🤷🏼

So, should the comment just be something like “TODO: `Err` is ignored to match the C++ behavior. In future, allow this operation to fail.”?

Reply:
"... In future, consider allowing this operation to fail." (I haven't checked whether that is consensus-compatible.)